pkg/scene/builder.go

```diff
@@ -48,57 +48,54 @@ func (b *Builder) MustBuild() *Scene {
 	return r
 }
 
-// ID _
 func (b *Builder) ID(id id.SceneID) *Builder {
 	b.scene.id = id
 	return b
 }
 
-// NewID _
 func (b *Builder) NewID() *Builder {
 	b.scene.id = id.SceneID(id.New())
 	return b
 }
 
-// Project _
 func (b *Builder) Project(prj id.ProjectID) *Builder {
 	b.scene.project = prj
 	return b
 }
 
-// Team _
 func (b *Builder) Team(team id.TeamID) *Builder {
 	b.scene.team = team
 	return b
 }
 
-// UpdatedAt _
 func (b *Builder) UpdatedAt(updatedAt time.Time) *Builder {
 	b.scene.updatedAt = updatedAt
 	return b
 }
 
-// WidgetSystem _
 func (b *Builder) WidgetSystem(widgetSystem *WidgetSystem) *Builder {
 	widgetSystem2 := *widgetSystem
 	b.scene.widgetSystem = &widgetSystem2
 	return b
 }
 
-// RootLayer _
+func (b *Builder) WidgetAlignSystem(widgetAlignSystem *WidgetAlignSystem) *Builder {
+	widgetAlignSystem2 := *widgetAlignSystem
+	b.scene.widgetAlignSystem = &widgetAlignSystem2
```

Comment:
You don't need to clone it because the contents of WidgetAlignSystem are hidden by private fields and methods. That's what "encapsulation" (OOP term) is for.

```suggestion
	b.scene.widgetAlignSystem = widgetAlignSystem
```

On the other hand, value objects whose fields are all public should be cloned. Otherwise, unintended scene changes will occur. But even in that case, this code is not safe to clone　when widgetAlignSystem has any references as children.